### PR TITLE
MacOS: Fix path to shared libraries from binary dependency list on install

### PR DESCRIPTION
**Pull request by geffrak**
### Linked issues

N/A

### Summarize your change.

When running the install step, make sure that there's no dependencies to files in the build directory. 

### Describe the reason for the change.

Fixes io_oiio.dylib (and libOpenImageIO.2.4.6.dylib), which contained hardcoded full paths to 
files in the build RV_DEPS area. 

### Describe what you have tested and on which operating system.

Tested on MacOS, where I built stuff using Autodesk's CI, tested on my MacBook running Sonoma.

### Add a list of changes and note any needing special attention during the review.

N/A

### If possible, could you provide screenshots

```
$ otool -L RV/RV.app/Contents/PlugIns/ImageFormats/io_oiio.dylib 
RV/RV.app/Contents/PlugIns/ImageFormats/io_oiio.dylib:
	@rpath/io_oiio.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libTwkFB.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libOpenEXR-3_1.30.dylib (compatibility version 30.0.0, current version 30.7.1)
	@rpath/libIex-3_1.30.dylib (compatibility version 30.0.0, current version 30.7.1)
	@rpath/libIlmThread-3_1.30.dylib (compatibility version 30.0.0, current version 30.7.1)
	@rpath/libz.1.dylib (compatibility version 1.0.0, current version 1.2.13)
	@rpath/libOpenImageIO.2.4.dylib (compatibility version 2.4.0, current version 2.4.6)
	@rpath/libOpenImageIO_Util.2.4.dylib (compatibility version 2.4.0, current version 2.4.6)
	@rpath/libTwkUtil.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libTwkExc.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libImath-3_1.29.dylib (compatibility version 29.0.0, current version 29.4.0)
	@rpath/libboost_thread.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.23.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)

$ otool -L RV/RV.app/Contents/lib/libOpenImageIO.2.4.6.dylib 
RV/RV.app/Contents/lib/libOpenImageIO.2.4.6.dylib:
	@rpath/libOpenImageIO.2.4.dylib (compatibility version 2.4.0, current version 2.4.6)
	@rpath/libOpenImageIO_Util.2.4.dylib (compatibility version 2.4.0, current version 2.4.6)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	@rpath/libz.1.dylib (compatibility version 1.0.0, current version 1.2.13)
	@rpath/libavcodec.58.dylib (compatibility version 58.0.0, current version 58.134.100)
	@rpath/libavformat.58.dylib (compatibility version 58.0.0, current version 58.76.100)
	@rpath/libavutil.56.dylib (compatibility version 56.0.0, current version 56.70.100)
	@rpath/libswscale.5.dylib (compatibility version 5.0.0, current version 5.9.100)
	/usr/lib/libbz2.1.0.dylib (compatibility version 1.0.0, current version 1.0.8)
	@rpath/libopenjp2.7.dylib (compatibility version 7.0.0, current version 2.5.0)
	@rpath/libOpenEXR-3_1.30.dylib (compatibility version 30.0.0, current version 30.7.1)
	@rpath/libOpenEXRCore-3_1.30.dylib (compatibility version 30.0.0, current version 30.7.1)
	@rpath/libraw_r.23.dylib (compatibility version 24.0.0, current version 24.0.0)
	@rpath/libtiff.5.dylib (compatibility version 5.0.0, current version 5.3.0)
	@rpath/libboost_thread.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libImath-3_1.29.dylib (compatibility version 29.0.0, current version 29.4.0)
	@rpath/libIlmThread-3_1.30.dylib (compatibility version 30.0.0, current version 30.7.1)
	@rpath/libIex-3_1.30.dylib (compatibility version 30.0.0, current version 30.7.1)
	@rpath/libjpeg.62.dylib (compatibility version 62.0.0, current version 62.3.0)
	@rpath/libboost_chrono.dylib (compatibility version 0.0.0, current version 0.0.0)
	@rpath/libboost_atomic.dylib (compatibility version 0.0.0, current version 0.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.23.0)
```

### CODE REVIEW

File: cmake/install/pre_install_darwin.cmake [hunk 1]
```diff
@@ -83,6 +83,7 @@ FUNCTION(after_copy_platform FILE_PATH)
     RETURN()
   ENDIF()
 
-  EXECUTE_PROCESS(COMMAND python3 ${CMAKE_CURRENT_LIST_DIR}/../../src/build/remove_absolute_rpath.py --target ${FILE_PATH} COMMAND_ERROR_IS_FATAL ANY)
+  MESSAGE(STATUS "python3 ${CMAKE_CURRENT_LIST_DIR}/../../src/build/remove_absolute_rpath.py --target ${FILE_PATH} --root ${CMAKE_SOURCE_DIR}")
```
Comment (geffrak) [inline cmake/install/pre_install_darwin.cmake:86]: Print the function being executed for debugging purposes.

File: cmake/install/pre_install_darwin.cmake [hunk 2]
```diff
@@ -83,6 +83,7 @@ FUNCTION(after_copy_platform FILE_PATH)
...
+  EXECUTE_PROCESS(COMMAND python3 ${CMAKE_CURRENT_LIST_DIR}/../../src/build/remove_absolute_rpath.py --target ${FILE_PATH} --root ${CMAKE_SOURCE_DIR} COMMAND_ERROR_IS_FATAL ANY)
```
Comment (geffrak) [inline cmake/install/pre_install_darwin.cmake:87]: Adds the --root to tell the script the boundaries of the convert to @rpath operation. Right now, it's set to the repository path.

File: src/build/remove_absolute_rpath.py [hunk 3]
```diff
@@ -24,29 +24,44 @@ def get_object_files(target):
 
                 file_type = subprocess.check_output(["file", "-bh", path])
                 if file_type.startswith(b"Mach-O"):
-                    yield path
+                    yield str(path.absolute())
```
Comment (geffrak) [inline src/build/remove_absolute_rpath.py:27]: return a string rather than a pathlib object

File: src/build/remove_absolute_rpath.py [hunk 4]
```diff
@@ -24,29 +24,44 @@ def get_object_files(target):
...
     else:
         yield target
 
 
 def get_rpaths(object_file_path):
     otool_output = subprocess.check_output(
         ["otool", "-l", object_file_path]
-    ).splitlines()
+    ).decode().splitlines()
```
Comment (geffrak) [inline src/build/remove_absolute_rpath.py:35]: convert bytes to str

File: src/build/remove_absolute_rpath.py [hunk 5]
```diff
@@ -24,29 +24,44 @@ def get_object_files(target):
...
+    otool_output = subprocess.check_output(
+        ["otool", "-L", object_file_path]
+    ).decode().splitlines()
+
+    i = 0
+
+    while i < len(otool_output):
+        otool_line = otool_output[i].strip()
+        i += 1
+        if "(compatibility version" in otool_line:
```
Comment (geffrak) [inline src/build/remove_absolute_rpath.py:63]: I'm not proud of this line, but it's working

File: src/build/remove_absolute_rpath.py [hunk 6]
```diff
@@ -57,29 +72,60 @@ def delete_rpath(object_file_path, rpath):
...
 
     args = parser.parse_args()
 
     if args.target.exists() is False:
         raise FileNotFoundError(f"Unable to locate {args.target.absolute()}")
+    if args.root.exists() is False:
+        raise FileNotFoundError(f"Unable to locate {args.root.absolute()}")
+
+    # fail if target is not in root
+    if args.target.resolve().relative_to(args.root.resolve()) is False:
```
Comment (geffrak) [inline src/build/remove_absolute_rpath.py:128]: Make sure the target is a leaf of the root.

**Review by rogernelson — APPROVED** [review]

**Review by bernie-laberge — APPROVED** [review]

**Pull request merged**